xdsl/transforms/experimental/convert_stencil_to_ll_mlir.py

```diff
@@ -497,6 +511,46 @@ def match_and_rewrite(self, op: ExternalStoreOp, rewriter: PatternRewriter, /):
         rewriter.erase_matched_op()
 
 
+class CombineOpCleanup(RewritePattern):
+    """
+    Just remove `stencil.combine`s as they are just used for return target analysis.
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: CombineOp, rewriter: PatternRewriter, /):
+        rewriter.erase_matched_op()
+
+
+def _get_use_target(use: Use) -> SSAValue | None:
+    match store := use.operation:
+        case StoreOp():
+            return store.field
+        case BufferOp():
+            return store.temp
+        case CombineOp():
+            # If the use is a lower
+            if use.index < len(store.lower):
+                temp = store.results[use.index]
+            elif use.index < len(store.lower) + len(store.upper):
+                temp = store.results[use.index - len(store.lower)]
+            elif use.index < len(store.lower) + len(store.upper) + len(store.lowerext):
+                temp = store.results[use.index - len(store.lower)]
+            else:
+                temp = store.results[use.index - len(store.lower) - len(store.lowerext)]
+            temp_uses = temp.uses
```

Comment:
Maybe add some short example docstring for this?

Reply:
Right !
This actually something I think we should have some core helpers for
Will document as workaround 🙂

Reply:
Probably just a `CombineOp` helper; I'll see if it needs reuse.